### ISSUE TITLE
Class mappings for ValueType enum should be mutable

### ISSUE THREAD

**Issue by aseovic** [owlike/genson#130]
While the default class mappings for `ValueType` enum make sense, there are cases when users may want to override them.

For example, we have our own class that all untyped JSON objects should map to, and a converter for it, but in order for the converter to kick in we need to be able to change mapping for the `ValueType.OBJECT` from `Map.class` to our class. 

In a more general case, it may be useful to use `javax.json.JsonObject` as a default for the `ValueType.OBJECT`. I'd even argue that should be the default, but considering that changing the default would break backwards compatibility in a major way, allowing users to change it is probably the best we can do.

Finally, while `ValueType.OBJECT` is probably the most common value users will want to change the mapping for, I can see how changing the mapping for `ValueType.DOUBLE` to `BigDecimal` can come in handy in some cases.

**Comment by EugenCepoi** [comment]
If someone wants to ser/de to other types when the static type is unknown the solution is to implement something like [UntypedConverterFactory](https://github.com/owlike/genson/blob/492868f21d1b3fb0de41c8a4e52f68c4544f4f4a/genson/src/main/java/com/owlike/genson/convert/DefaultConverters.java#L983-L1014).

Isn't this enough?

**Comment by aseovic** [comment]
Sure, that's an option, but it seems like an overkill when a simple remapping would do.

Implementing and registering custom converter is significantly more involved, so having an out-of-the box way to define the mappings from ValueType -> class that should be used for deserialization would be nice.

**Comment by aseovic** [comment]
Fair enough, I agree that making `ValueType` mutable is a bit hacky, but it was the quickest way to achieve the immediate goal ;-)

Let me re-work it and implement a cleaner way to modify the mappings and I'll resubmit the PR.
